apps/image-server/server.js

```diff
@@ -155,26 +150,36 @@ app.get('/image/*',
  *  The url for creating one Image
  */
 app.post('/image',
-  passport.authenticate('bearer', {session: false}),
   upload.single('image'), (req, res, next) => {
     // req.file is the `image` file
     // req.body will hold the text fields, if there were any
     //
-    if (!res.headerSent) {
-      res.setHeader('Content-Type', 'application/json');
+    const verification = crypto.createHmac("sha256", secret).digest("hex")
+    if(Date.now() < req.query.exp_date && verification === req.query.signature) {
```

Comment:
Ik kan nu met de hand de exp_date aanpassen, en dan wordt dat hier goed gekeurd. Dat lijkt me niet de bedoeling. Ik denk dat de exp_date deel moet uitmaken van het ge-encrypte token. Anders kan ik een eenmaal afgegeven token tot in de eeuwigheid blijven gebruiken.

Reply:
Misschien is het handig om dan jwt's te gebruiken ipv. crypto. De libs zijn er al, dus geen extra afhankelijkheden, en exp_date zit daar al ingebakken.

Reply:
Klopt inderdaad. Mijn alternatieve voorstel is om de secret key en de exp_date met elkaar te combineren. Op die manier is de exp_date best te vervalsen, maar dan kan je ook niet langer de url verifiëren.
`const createdCombination = secret + req.query.exp_date`
`const verification = crypto.createHmac("sha256", createdCombination).digest("hex")`